.github/workflows/pr-preview.yml

```
@@ -0,0 +1,53 @@
+name: PR Preview
+
+on:
+  pull_request:
+    types: [opened, reopened, synchronize, closed]
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  preview:
+    runs-on: ubuntu-latest
+
+    concurrency:
+      group: preview-${{ github.repository }}-${{ github.event.pull_request.number }}
+      cancel-in-progress: true
+
+    steps:
+      - name: Checkout source repo
+        uses: actions/checkout@v4
+
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 22
+
+      - name: Build with PR-specific baseUrl
+        if: github.event.action != 'closed'
+        working-directory: website
+        run: |
+          set -euo pipefail
+          PR_PATH="/pr-preview/spec-toolkit/pr-${{ github.event.pull_request.number }}/"
+          CFG="docusaurus.config.js"
+
+          cp "$CFG" "$CFG.bak"
+          trap 'mv -f "$CFG.bak" "$CFG"' EXIT
+
+          sed -i 's#baseUrl: "/spec-toolkit/",#baseUrl: "'"$PR_PATH"'",#' "$CFG"
+          grep -q "baseUrl: \"$PR_PATH\"" "$CFG"
+
+          npm ci
+          npm run build
+
+      - name: Deploy / Remove preview in central repo
+        uses: rossjrw/pr-preview-action@v1
+        with:
+          source-dir: website/build
+          deploy-repository: open-resource-discovery/pr-preview
+          preview-branch: main
+          umbrella-dir: spec-toolkit
+          pages-base-url: open-resource-discovery.github.io/pr-preview
+          token: ${{ secrets.PR_PREVIEW_DEPLOY_TOKEN }}
```

```suggestion
          token: ${{ secrets.PR_PREVIEW_TOKEN }}
```

This is a global secret in our organization, it's available to all repos.